cmake/scripts/ethupdate.sh

```diff
@@ -5,20 +5,81 @@
 # Invoke from the root directory and make sure you have the arguments set as explained
 # in the usage string.
 
+
+# Get SCRIPT_DIR, the directory the script is located even if there are symlinks involved
+FILE_SOURCE="${BASH_SOURCE[0]}"
+# resolve $FILE_SOURCE until the file is no longer a symlink
+while [ -h "$FILE_SOURCE" ]; do
+	SCRIPT_DIR="$( cd -P "$( dirname "$FILE_SOURCE" )" && pwd )"
+	FILE_SOURCE="$(readlink "$FILE_SOURCE")"
+	# if $FILE_SOURCE was a relative symlink, we need to resolve it relative to the path where the symlink file was located
+	[[ $FILE_SOURCE != /* ]] && FILE_SOURCE="$SCRIPT_DIR/$FILE_SOURCE"
+done
+SCRIPT_DIR="$( cd -P "$( dirname "$FILE_SOURCE" )" && pwd )"
+
+# Now that we got the directory, source some common functionality
+source "${SCRIPT_DIR}/ethbuildcommon.sh"
+
 ROOT_DIR=$(pwd)
-PROJECTS=(cpp-ethereum cpp-ethereum-cmake webthree solidity alethzero mix)
+NO_PUSH=0
+USE_SSH=0
 UPSTREAM=upstream
 ORIGIN=origin
 REQUESTED_BRANCH=develop
 REQUESTED_ARG=""
+REQUESTED_PROJECT=""
+REPO_URL=""
+REPOS_MAP=("webthree-helpers:https://github.com/ethereum/webthree-helpers"
+	   "tests:https://github.com/ethereum/tests"
+	   "libweb3core:https://github.com/ethereum/libweb3core"
+	   "libethereum:https://github.com/ethereum/libethereum"
+	   "libwhisper:https://github.com/ethereum/libwhisper"
+	   "webthree:https://github.com/ethereum/webthree"
+	   "solidity:https://github.com/ethereum/solidity"
+	   "alethzero:https://github.com/ethereum/alethzero"
+	   "mix:https://github.com/ethereum/mix")
+REPOS_SSH_MAP=("webthree-helpers:git@github.com:ethereum/webthree-helpers.git"
+	   "tests:git@github.com:ethereum/tests.git"
+	   "libweb3core:git@github.com:ethereum/libweb3core.git"
+	   "libethereum:git@github.com:ethereum/libethereum.git"
+	   "libwhisper:git@github.com:ethereum/libwhisper.git"
+	   "webthree:git@github.com:ethereum/webthree.git"
+	   "solidity:git@github.com:ethereum/solidity.git"
+	   "alethzero:git@github.com:ethereum/alethzero.git"
+	   "mix:git@github.com:ethereum/mix.git")
```

Comment:
I think that you also need to pull `web.js` submodule to build webthree (eth), alethzero and mix